src/ReactiveDomain/ReactiveDomain.Tests/Asserts/Sdk/ArgumentFormatter.cs

```diff
@@ -58,62 +59,96 @@ static string Format(object value, int depth)
+            try
             {
-                var typeParameters = typeInfo.GenericTypeArguments;
-                var typeName = typeParameters.Length == 0 ? "Task" : string.Format("Task<{0}>", string.Join(",", typeParameters.Select(FormatTypeName)));
-                return string.Format("{0} {{ Status = {1} }}", typeName, task.Status);
-            }
-
-#if PLATFORM_DOTNET
-            var toString = type.GetRuntimeMethod("ToString", EmptyTypes);
-#else
-            var toString = type.GetMethod("ToString", EmptyTypes);
-#endif
-
-            if (toString != null && toString.DeclaringType != typeof(Object))
-                return (string)toString.Invoke(value, EmptyObjects);
-
-            return FormatComplexValue(value, depth, type);
+                if (value.GetType().GetTypeInfo().IsEnum)
+                    return value.ToString().Replace(", ", " | ");
+
+                if (value is char)
+                {
+                    var charValue = (char)value;
+                
+                    if (charValue == '\'')
+                        return @"'\''";
+                
+                    // Take care of all of the escape sequences
+                    string escapeSequence;
+                    if (TryGetEscapeSequence(charValue, out escapeSequence))
+                    {
+                        return $"'{escapeSequence}'";
+                    }
+                
+                    if (char.IsLetterOrDigit(charValue) || char.IsPunctuation(charValue) || char.IsSymbol(charValue) || charValue == ' ')
+                        return $"'{charValue}'";
+
+                    // Fallback to hex
+                    return $"0x{(int)charValue:x4}";
+                }
+
+                if (value is DateTime || value is DateTimeOffset)
+                    return $"{value:o}";
+
+                var stringParameter = value as string;
+                if (stringParameter != null)
+                {
+                    stringParameter = EscapeHexChars(stringParameter);
+                    stringParameter = stringParameter.Replace(@"""", @"\"""); // escape double quotes
+                    if (stringParameter.Length > MAX_STRING_LENGTH)
+                    {
+                        string displayed = stringParameter.Substring(0, MAX_STRING_LENGTH);
+                        return $"\"{displayed}\"...";
+                    }
+
+                    return $"\"{stringParameter}\"";
+                }
+
+                try
+                {
+                    var enumerable = value as IEnumerable;
+                    if (enumerable != null)
+                        return FormatEnumerable(enumerable.Cast<object>(), depth);
+                }
+                catch
+                {
+                    // Sometimes enumerables cannot be enumerated when being, and instead thrown an exception.
+                    // This could be, for example, because they require state that is not provided by Xunit.
```

Comment:
As I said in the pull request description, "You can ignore changes to all the files under ReactiveDomain.Tests/Asserts/ since they're pulled in automatically by the xunit.assert.source nuget."